### ISSUE TITLE
Duplicate plot entries

### ISSUE THREAD

**Issue by aaronevers** [jneilliii/OctoPrint-PlotlyTempGraph#53]
I occasionally get duplicate plot entries. 

![Screenshot from 2024-07-05 11-24-25](https://github.com/jneilliii/OctoPrint-PlotlyTempGraph/assets/13547569/1cf03cc1-4ca6-47de-96ac-c4e2b95ad8cb)

**Comment by jneilliii** [comment]
Not 100% sure what could cause this to happen. Looks like you're running Prusa firmware based on what you have listed. If you look in settings do those additional duplicates also show to be able to set their properties or hide?

**Comment by aaronevers** [comment]
Octoprint: 1.20.2; Plotly 0.1.7

Prusa MK3s, firmware: 3.13.3. Happens with both a MMU instance of Octoprint and a separate instance without it. Seems to happen at the same time to both. I haven't been able to determine if it's something in the terminal causing it.

I see no duplicate entries in the Name Mapping table.

**Comment by jneilliii** [comment]
That's very weird. I'd be interested in seeing if there are any errors in the browser's developer tools > console tab when this happens. To determine if it's something related to the serial communication you could enable serial logging and then we can check out serial.log to see if there's anything that stands out where it may be reporting these duplicates. I have a sneaky suspicion that this may be a hard one to track down though due to some random race condition.

**Comment by aaronevers** [comment]
![Screenshot from 2024-07-05 23-55-43](https://github.com/jneilliii/OctoPrint-PlotlyTempGraph/assets/13547569/52e274a5-ae15-4307-86c3-cfd6f66aa315)
[serial.log.zip](https://github.com/user-attachments/files/16114908/serial.log.zip)

Left the log going until I saw duplicates.  Printer was idle.

**Comment by jneilliii** [comment]
Yeah, nothing in the serial.log looks out of place. Could be related to that websocket reconnection. It's doing plotTraces, and then when reconnects merged in fromHistoryData and then continues with plotTraces. The fromHistoryData should be all the previous plotTraces, but there could be some weird merging issue. Could be an upstream plotly bug, and it's been a while since I've updated the library.
